content/chapters/io/lecture/slides.mdpp

```diff
@@ -13,22 +13,12 @@ revealOptions:
 1. Interprocess Communication
 1. IO Optimizations
 
----
-
 !INCLUDE "slides/intro.md"
```

Comment:
Moving the delimiters to `.md` triggers `MD041`. I think it would be better to leave this as it was. If someone would use only a `.md` file it would have to manually remove the starting delimiters.

Reply:
I agree with @Alex-deVis.

Reply:
@Alex-deVis, please elaborate on triggering `MD041` by moving delimiters. The starting of a Markdown page in the `slides/` directory is now often `##` (level 2), which is OK; it's too much to have it `#` (level 1).

How does moving delimiters help with this?

Also, delimiters are already part of the Markdown files in `slides/`. Why would the keeping of the first delimiters in `slides.mdpp` file help much with that? For starters, the `.mdpp` file would delimiter-free.

Reply:
So `MD041` simply asks that all `.md` files start with `#`. So if you place delimiters inside the `.md` files, they are guaranteed to fail `MD041`. Our proposal is to keep the `.md` files standalone, which implies 2 things:
1. place the delimiters inside the `.mdpp` file. This is OK since that is mostly an aggregator for files containing the actual content
2. Start each `.md` file with a `#`. This may be too much, but since they usually start with a title-only slide, this should be fine.

Keep in mind, though that this change will create a ripple effect because `markdownlint` also enforces that a given heading be at most 1 `#` lower than its parent. So if the initial heading is `#`, then the second one must be `##` and so on. I know there are some slides in which I could barely fit the code / bullet points using `####`. Making them `###` will lead to spillovers and to the need to rearrange slides.

Writing all of this, I'm beginning to that think abiding by `MD041` is more trouble than it's worth. But at least we should move the initial delimiters from the `.md` files to the `.mdpp` file as this would make the Markdown files slightly more modular and easy to pick up by other potential teachers.

Reply:
@teodutu, @Alex-deVis, I agree with dropping `MD041`. It's too cumbersome. For the placing of delimiters, OK, I'll yield and have them as part of `.mdpp`. I will create a PR to make this uniform across all sections.